linalg_ops/convolution/generate_e2e_conv2d_tests.py

```diff
@@ -0,0 +1,739 @@
+#!/usr/bin/env python3
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Generator for e2e conv2d tests.
+"""
+
+import argparse
+import enum
+import dataclasses
+import typing
+import math
+
+
+# Data type of kernel entries. The string values must match MLIR data types.
+@enum.unique
+class KernelElemTypeId(enum.Enum):
+    NONE = ""
+    I8 = "i8"
+    F32 = "f32"
+    F16 = "f16"
+
+
+# Data type of input entries. The string values must match MLIR data types.
+@enum.unique
+class InputElemTypeId(enum.Enum):
+    NONE = ""
+    I8 = "i8"
+    F32 = "f32"
+    F16 = "f16"
+
+
+# Data type of input entries. The string values must match MLIR data types.
+@enum.unique
+class AccElemTypeId(enum.Enum):
+    NONE = ""
+    I32 = "i32"
+    F32 = "f32"
+    F16 = "f16"
+
+
+# Enumerates of the collections of shapes that we can generate tests for.
+# The values are the accepted values for the --shapes= flag.
+@enum.unique
+class ShapesId(enum.Enum):
+    SMALL = "small"
+    MEDIUM = "medium"
+    LARGE = "large"
+    GPU_LARGE = "gpu_large"
```

Comment:
I'm removing the concept of "GPU shapes" from the matmul tests in https://github.com/iree-org/iree-test-suites/pull/22. See the discussion thread at https://github.com/iree-org/iree-test-suites/pull/22#discussion_r1747365789. Probably want to do the same here.